### PR TITLE
lexend: init at 0.pre+date=2022-01-27

### PR DESCRIPTION
**Pull request by fufexan**
###### Description of changes
Add Lexend font. https://www.lexend.com

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/fonts/lexend/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ lib, stdenvNoCC, fetchFromGitHub }:
+
+let
+  version = "0.pre+date=2022-01-27";
+  pname = "lexend";
```
Comment (legendofmiracles) [inline pkgs/data/fonts/lexend/default.nix:5]: Why? `rec` exists. And You only use version once anyways

File: pkgs/data/fonts/lexend/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      install $f/ttf/* $out/share/fonts/truetype/lexend/$f/
+    done
+  '';
+
+  meta = {
+    homepage = "https://www.lexend.com";
+    description = "A variable font family designed to aid in reading proficiency.";
+    license = lib.licenses.ofl;
+    platforms = lib.platforms.all;
+    maintainers = [ lib.maintainers.fufexan ];
```
Comment (legendofmiracles) [inline pkgs/data/fonts/lexend/default.nix:30]:
```suggestion
  meta = with lib; {
    homepage = "https://www.lexend.com";
    description = "A variable font family designed to aid in reading proficiency";
    license = licenses.ofl;
    platforms = platforms.all;
    maintainers = with maintainers; [ fufexan ];
```

File: pkgs/data/fonts/lexend/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    rev = "57e6c14e2a9b457e8376044a31525c2100297e9c";
+    sha256 = "sha256-+tPggQIO50a8kOSnOVN/MR9ZwX5xZqYVNZO79eog9QA=";
+  };
+
+  installPhase = ''
+    cd fonts
+    for f in *; do
+      mkdir -p $out/share/fonts/truetype/lexend/$f
+      install $f/ttf/* $out/share/fonts/truetype/lexend/$f/
+    done
```
Comment (legendofmiracles) [inline pkgs/data/fonts/lexend/default.nix:22]:
```suggestion
  installPhase = ''
    runPhase preInstall
    
    cd fonts
    for f in *; do
      mkdir -p $out/share/fonts/truetype/lexend/$f
      install $f/ttf/* $out/share/fonts/truetype/lexend/$f/
    done
    
    runPhase postInstall
```

**Review by legendofmiracles — CHANGES_REQUESTED** [review]

File: pkgs/data/fonts/lexend/default.nix (hunk 3 above)
Comment (fufexan) [reply pkgs/data/fonts/lexend/default.nix:22]: Oops forgot about pre/post.

File: pkgs/data/fonts/lexend/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitHub
+}:
+
+stdenvNoCC.mkDerivation rec {
+  version = "0.pre+date=2022-01-27";
+  pname = "lexend";
```
Comment (legendofmiracles) [inline pkgs/data/fonts/lexend/default.nix:8]:
```suggestion
  pname = "lexend";
  version = "0.pre+date=2022-01-27";
```
pname typically goes first

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 178231` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>lexend</li>
  </ul>
</details>

File: pkgs/data/fonts/lexend/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  };
+
+  installPhase = ''
+    runHook preInstall
+
+    cd fonts
+    for f in *; do
+      mkdir -p $out/share/fonts/truetype/lexend/$f
+      install $f/ttf/* $out/share/fonts/truetype/lexend/$f/
+    done
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/lexend/default.nix:24]: If we only copy files we can also move this to postFetch in fetchFromGitHub.

Comment (fufexan) [reply pkgs/data/fonts/lexend/default.nix:24]: I don't think it's working properly, or I'm missing something.
```nix
{ lib
, fetchFromGitHub
}:
let
  pname = "lexend";
  version = "0.pre+date=2022-01-27";
  rev = "57e6c14e2a9b457e8376044a31525c2100297e9c";
in
fetchFromGitHub {
  name = "${pname}-${version}";

  owner = "googlefonts";
  repo = pname;
  inherit rev;
  sha256 = "sha256-p1Yc+mM/felZuhpSrzltLdMcFG9ZP5riW3QmAYRFAf4=";

  postFetch = ''
    tar -xf $downloadedFile --strip=1
    mkdir -p $out/share/{fonts/truetype,doc}/lexend

    for f in fonts/*; do
      mkdir -p $out/share/fonts/truetype/lexend/$f
      cp -r fonts/$f/ttf/* $out/share/fonts/truetype/lexend/$f/
    done

    cp README.md documentation $out/share/doc/lexend
  '';

  meta = with lib; {
    homepage = "https://www.lexend.com";
    description = "A variable font family designed to aid in reading proficiency";
    license = licenses.ofl;
    platforms = platforms.all;
    maintainers = with maintainers; [ fufexan ];
  };
}
```

Comment (SuperSandro2000) [reply pkgs/data/fonts/lexend/default.nix:24]: no idea

**Comment by MonAaraj** [comment]
any progress on this? it seems like any development was paused for about a month, what're the blockers?

**Comment by fufexan** [comment]
I have tried to get it to work with `fetchFromGitHub` only, but due to the structure of the files it's not feasible. So I propose we merge it as-is, since we know it works.

Thoughts @SuperSandro2000 ?

**Comment by MonAaraj** [comment]
Yeah, I agree. Since no one in here knows how to get `fetchFromGithub` working properly, I think it is plenty good as it is.

**Pull request merged**